include/utility/vpd_specific_utility.hpp

```diff
@@ -193,7 +193,10 @@ inline void insertOrMerge(types::InterfaceMap& map,
     if (map.find(interface) != map.end())
     {
         auto& prop = map.at(interface);
-        prop.insert(propertyMap.begin(), propertyMap.end());
+        std::for_each(propertyMap.begin(), propertyMap.end(),
+                      [&prop](auto l_keyValue) {
+            prop[l_keyValue.first] = l_keyValue.second;
```

Comment:
have a concern on this line 
from the documentation I see the below, could this cause an issue on the for_each iterator ?
If the function inserts a new element, concurrently iterating ranges in the container is not safe.

Reply:
I guess it should not

Reply:
what about exception ?

Reply:
handled the exception.